substrate/frame/utility/src/lib.rs

```diff
@@ -454,6 +462,102 @@ pub mod pallet {
 			let res = call.dispatch_bypass_filter(frame_system::RawOrigin::Root.into());
 			res.map(|_| ()).map_err(|e| e.error)
 		}
+
+		/// Dispatch a fallback call in the event the main call fails to execute.
+		/// May be called from any origin except `None`.
+		///
+		/// This function first attempts to dispatch the `main` call.
+		/// If the `main` call fails, the `fallback` is attemted.
+		/// if the fallback is successfully dispatched, the weights of both calls
+		/// are accumulated and an event containing the main call error is deposited.
+		///
+		/// In the event of a fallback failure the whole call fails
+		/// with the weights returned.
+		///
+		/// - `main`: The main call to be dispatched. This is the primary action to execute.
+		/// - `fallback`: The fallback call to be dispatched in case the `main` call fails.
+		///
+		/// ## Dispatch Logic
+		/// - If the origin is `root`, both the main and fallback calls are executed without
+		///   applying any origin filters.
+		/// - If the origin is not `root`, the origin filter is applied to both the `main` and
+		///   `fallback` calls.
+		///
+		/// ## Use Case
+		/// - Some use cases might involve submitting a `batch` type call in either main, fallback
+		///   or both.
+		///
+		/// ## Weight
+		/// The weight of this call is calculated as the sum of:
+		/// `WeightInfo::if_else()` + `main` |
+		/// `WeightInfo::if_else()` + `main` + `fallback`.
+		#[pallet::call_index(6)]
+		#[pallet::weight({
+			let main = main.get_dispatch_info();
+			let fallback = fallback.get_dispatch_info();
+			(
+				T::WeightInfo::if_else()
+					.saturating_add(main.call_weight)
+					.saturating_add(fallback.call_weight),
+				if main.class == Operational && fallback.class == Operational { Operational } else { Normal },
+			)
+		})]
+		pub fn if_else(
+			origin: OriginFor<T>,
+			main: Box<<T as Config>::RuntimeCall>,
+			fallback: Box<<T as Config>::RuntimeCall>,
+		) -> DispatchResultWithPostInfo {
+			// Do not allow the `None` origin.
+			if ensure_none(origin.clone()).is_ok() {
+				return Err(BadOrigin.into());
+			}
+
+			let is_root = ensure_root(origin.clone()).is_ok();
+
+			// Track the weights
+			let mut weight = T::WeightInfo::if_else();
+
+			let info = main.get_dispatch_info();
+
+			// Execute the main call first
+			let main_result = if is_root {
+				main.dispatch_bypass_filter(origin.clone())
+			} else {
+				main.dispatch(origin.clone())
+			};
+
+			// Add weight of the main call
+			weight = weight.saturating_add(extract_actual_weight(&main_result, &info));
+
+			if let Err(main_error) = main_result {
```

Comment:
`let .. else` could remove the nesting. But its a very minor comment, feel free to ignore 😄

Reply:
Yeah, makes for a much cleaner code, thanks for the tip